static/js/auth.ts

```diff
@@ -150,90 +127,98 @@ export const auth = {
 
         afterLogin();
       }).fail (function (response) {
-        if (response.status >= 500) return auth.error (auth.texts['server_error']);
+        $ ('#create-account-button').hide ();
+        if (response.status >= 500) {
+           return auth.error (auth.texts['server_error']);
+        } else if (response.status == 400) {
+          auth.error (response.responseText);
+        }
         if (response.status === 403) {
-           auth.error (auth.texts['invalid_username_password'] + ' ' + auth.texts['no_account'] + ' &nbsp;<button class="green-btn" onclick="auth.redirect (\'signup\')">' + auth.texts['create_account'] + '</button>');
-           $ ('#create-account').hide ();
+           auth.error (auth.texts['invalid_username_password'] + ' ' + auth.texts['no_account']);
+           $ ('#create-account-button').show ();
            localStorage.setItem ('hedy-login-username', values.username ?? '');
+        } else {
+          auth.error (auth.texts['ajax_error']);
         }
-        else auth.error (auth.texts['ajax_error']);
       });
     }
 
     if (op === 'profile') {
-      if (! (values.email ?? '').match (auth.emailRegex)) return auth.error (auth.texts['valid_email'], 'email');
-
-      if (values.birth_year) {
-        if (!validBirthYearString(values.birth_year)) {
-          return auth.error (auth.texts['valid_year'] + new Date ().getFullYear (), 'birth_year');
-        }
-      }
-
       const payload: User = {
-        email: values['email'],
-        birth_year: values.birth_year ? parseInt(values['birth_year']) : undefined,
-        country: values['country'] ? values.country : undefined,
-        gender: values['gender'] ? values.gender : undefined,
+        email: values.email,
         language: values.language,
-        prog_experience: $ ('input[name=has_experience]:checked').val() as 'yes' | 'no',
+        birth_year: values.birth_year ? parseInt(values.birth_year) : undefined,
+        country: values.country ? values.country : undefined,
+        gender: values.gender ? values.gender : undefined,
+        prog_experience: $('input[name=has_experience]:checked').val() as 'yes'|'no',
         experience_languages: $('#languages').is(':visible')
           ? $('input[name=languages]').filter(':checked').map((_, box) => $(box).val() as string).get()
           : undefined,
       };
 
-      console.log(payload);
-
-      auth.clear_error ();
-      $.ajax ({type: 'POST', url: '/profile', data: JSON.stringify (payload), contentType: 'application/json; charset=utf-8'}).done (function () {
+      $.ajax ({
+        type: 'POST', url: '/profile',
+        data: JSON.stringify (payload),
+        contentType: 'application/json; charset=utf-8'
+      }).done (function () {
         auth.success (auth.texts['profile_updated']);
-        setTimeout (function () {location.reload ()}, 500);
+        setTimeout (function () {location.reload ()}, 1000);
```

Comment:
I thought that if the timeout was a bit longer users would be able to read the "success" message before the screens reloads. The reason we reload is because if a users changes their preferred language we want to update the complete UI, including the menubar etc. As the language is stored in the session the easiest way to do this is through a reload. 

It might be best to improve this implementation to only reload if the preferred language is changed. We can simply add this as a boolean to the result of the back-end. Will take a look!

Reply:
A second is enough if you know where to look, but not all users might know that.

Another solution is to persist the success message in a place where it would survive a reload.

Reply:
Just changed it to `1500`. I agree that this is not an ideal approach, but it should be suitable for now. I will open an issue to think about a more elegant way to implement the feedback and reload the page.